bin/wasm-node/javascript/src/index.js

```diff
@@ -79,11 +106,40 @@ export async function start(config) {
       // `expected` was pushed by the `addChain` method.
       // Resolve the promise that `addChain` returned to the user.
       expected.resolve({
+        health: () => {
+          // are we already waiting for a health message reply?
+          if (chainHealthRequestCallbacks[chainId]) {
+            const reject = chainHealthRequestCallbacks[chainId].reject;
+            delete chainHealthRequestCallbacks[chainId];
+            return reject(new Error('Called health before pending health request replied'));
```

Comment:
This feels very arbitrary. Why would it be forbidden to call `health()` multiple times in parallel?

Reply:
Sure but why would you make multiple health requests to the same chain in parallel? How about replacing the request when subsequent requests are made instead? Tracking multiple concurrent requests adds more complexity

Reply:
"why would anyone use this API in a way that is different from the way I use it" is generally what leads to bugs and clusterfucks.

Reply:
yeah you're right :+1: now I think about it not tracking all the requests is also a bad idea because users will end up with pending (never to be resolved) promises

Reply:
Done